superduperdb/ext/torch/model.py

```diff
@@ -319,22 +342,20 @@ def __post_init__(self):
         self._validation_set_cache = {}
 
     @cached_property
-    def optimizer(self):
+    def optimizer(self) -> Artifact:
         return self.training_configuration.optimizer_cls.artifact(
             self.object.artifact.parameters(),
             **self.training_configuration.optimizer_kwargs,
         )
 
     @property
-    def optimizers(self):
+    def optimizers(self) -> t.List[Artifact]:
         return [self.optimizer]
 
-    def save(self, database: DB):
-        self.optimizer_state = Artifact(self.optimizer.state_dict(), serializer='torch')
```

Comment:
The oopsie is here: `self.optimizer.state_dict()` doesn't exist, it should be `self.state_dict()`.